tfhe/src/core_crypto/commons/ciphertext_modulus.rs

```diff
@@ -31,6 +31,13 @@ pub struct CiphertextModulus<Scalar: UnsignedInteger> {
     _scalar: PhantomData<Scalar>,
 }
 
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub enum CiphertextModulusKind {
+    Native,
+    NonNativePowerOfTwo,
+    NonNative,
+}
```

Comment:
`CiphertextModulusInner` could have these 3 variants

Reply:
no, as then the ciphertext modulus is not a u128 anymore